packages/core/src/codewhisperer/models/errors.ts

```diff
@@ -0,0 +1,117 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { ToolkitError } from '../../shared/errors'
+import {
+    DefaultCodeScanErrorMessage,
+    FileSizeExceededErrorMessage,
+    NoWorkspaceFoundErrorMessage,
+    ProjectSizeExceededErrorMessage,
+    InvalidSourceFilesErrorMessage,
+    UploadArtifactToS3ErrorMessage,
+} from './constants'
+
+export enum ErrorCodes {
+    FileSizeExceeded = 'FileSizeExceeded',
+    ProjectSizeExceeded = 'ProjectSizeExceeded',
+    NoWorkspaceFound = 'NoWorkspaceFound',
+    InvalidSourceFiles = 'InvalidSourceFiles',
+    CreateUploadUrlError = 'CreateUploadUrlError',
+    UploadArtifactToS3Error = 'UploadArtifactToS3Error',
+    DefaultError = 'DefaultError',
+    InvalidSourceZip = 'InvalidSourceZip',
+    CreateCodeScanError = 'CreateCodeScanError',
+    CreateCodeScanFailedError = 'CreateCodeScanFailedError',
+    SecurityScanTimedOutError = 'SecurityScanTimedOutError',
+    CodeScanJobFailedError = 'CodeScanJobFailedError',
+}
+
+export const mapErrorToCustomerFacingMessage: Record<ErrorCodes, string> = {
+    FileSizeExceeded: FileSizeExceededErrorMessage,
+    ProjectSizeExceeded: ProjectSizeExceededErrorMessage,
+    NoWorkspaceFound: NoWorkspaceFoundErrorMessage,
+    InvalidSourceFiles: InvalidSourceFilesErrorMessage,
+    CreateUploadUrlError: DefaultCodeScanErrorMessage,
+    UploadArtifactToS3Error: UploadArtifactToS3ErrorMessage,
+    DefaultError: DefaultCodeScanErrorMessage,
+    InvalidSourceZip: DefaultCodeScanErrorMessage,
+    CreateCodeScanError: DefaultCodeScanErrorMessage,
+    CreateCodeScanFailedError: DefaultCodeScanErrorMessage,
+    SecurityScanTimedOutError: DefaultCodeScanErrorMessage,
+    CodeScanJobFailedError: DefaultCodeScanErrorMessage,
+}
+
+export class FileSizeExceededError extends ToolkitError {
+    constructor() {
+        super('Payload size limit reached', {
+            code: ErrorCodes.FileSizeExceeded,
+        })
+    }
+}
+
+export class ProjectSizeExceededError extends ToolkitError {
```

Comment:
you are creating custom classes for every error, yet not using the features of a class. the class could store the "customer facing message", and possibly could also avoid the enums and "mapping" indirection

Reply:
Created a new `SecurityScanError` class which extends `ToolkitError` to add a new field `customerFacingMessage`. This should work without `ErrorCodes` enum and `mapErrorToCustomerFacingMessage`.